GLOSSARY.md

```diff
@@ -0,0 +1,29 @@
+1. kept sprite - A sprite that was in the DOM in the previous render, and is not scheduled for removal in this render. Should have initial and final DOM state. may have a counterpart.
+2. inserted sprite - A sprite that was not in the DOM in the previous render, and was inserted/is present in the DOM in this render. Should have final DOM state, but not initial.
+3. removed sprite - A sprite that was in the DOM in the previous render, and is no longer in/is removed from the DOM this render. Should have initial DOM state, but not final.
+4. intermediate sprite - Sprites created from sprite modifiers that were registered in the previous render. They are used to identify possible interruptions of sprite's animations, and to transfer animated state like position from one render to the next (how?). They are a separate category from kept sprites, inserted sprites, and removed sprites.
+6. counterpart - when a kept sprite is created as a result of a different DOM element (eg via an if helper) being rendered with the same modifier id and role, the old element and its DOM state are used as the counterpart.
+7. sprite modifier - an Ember modifier that registers a `SpriteModifier` when the DOM element it is attached to is rendered, and unregisters the `SpriteModifier` when the DOM element it is attached to is removed from the DOM. The `SpriteModifier` has information about: 
+	- the modifier's DOM element
+	- the current state (eg the bounds) of the DOM element
+	- the id and role that a sprite created from it will take on
+8. sprite - an object that stores DOM state (whether initial, final, or both, depending on the type) so that it can be animated. Can be kept, inserted, removed, or intermediate.
+9. a sprite's role - a way of identifying a sprite. does not have to be unique
+10. a sprite's id - a unique way of identifying a sprite. 
```

Comment:
A sprite's ID can be the same as another sprite if the other sprite represents the same model, right? As in a list item and an expanded rendering?

Reply:
Yes, sprites can have the same identifier while they are being processed. My understanding is that we should not have duplicate sprite identifiers once we get to the transition function that the user passes to the `AnimationContext`, we reconcile duplicate identifiers (eg. sprites that are removed and re-inserted become kept sprites) and/or assert for them in certain ways to prevent this from happening.